File: src/main/java/com/vielheit/core/repository/UserRepository.java
package com.vielheit.core.repository;

import com.vielheit.core.domain.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import javax.transaction.Transactional;
import java.util.List;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    List<User> findByEmailAddress(String email);
}

File: src/main/kotlin/org/vielheit/core/repository/UserRepository.kt
package org.vielheit.core.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository
import org.vielheit.core.domain.User
import java.util.*

@Repository
interface UserRepository : JpaRepository<User, Int> {
    fun findByEmail(email: String): User?

    @Query("""SELECT COUNT(u.id) FROM User u WHERE u.email=:email""")
    fun countByEmail(email: String): Int
}